docs/topics/projections.rst

```diff
@@ -467,25 +471,25 @@ notification log even when the notification doesn't imply a real entry in the in
                     )
                     index_key = uuid_from_url(event.email_address)
                     index_value = event.originator_id
-                else:
-                    index_key = uuid.uuid4()
-                    index_value = ''
-
-                # Todo: And if we can't create new index item, get existing and append value.
-                index_item = IndexItem.__create__(originator_id=index_key, index_value=index_value)
-                index_item.__save__()
+                    try:
+                        index_item = self.repository[index_key]
+                        index_item.update_index(index_value)
+                    except KeyError:
+                        index_item = IndexItem.__create__(originator_id=index_key, index_value=index_value)
+                    index_item.__save__()
 
 
     # Construct original application.
     original = SQLAlchemyApplication(persist_event_type=User.Event)
 
     # Construct index application.
-    index = SQLAlchemyApplication(persist_event_type=IndexItem.Event)
+    index = SQLAlchemyApplication(name='indexer', persist_event_type=IndexItem.Event)
```

Comment:
I have added name to the application to make Indexer recoverable.
Now if i remove the projection from event store. On restart indexer will get 0 checkpoint. And will re-poroject. 

I also wonder shouldnt `persist_event_type=IndexItem.Event` be a query filter clause as well when retreiving `max_record_id`.